src/planwise/client/current_project/subs.cljs

```diff
@@ -96,3 +135,47 @@
  (fn [db [_]]
    (let [current-region-id (reaction (get-in @db [:current-project :project-data :region-id]))]
      (reaction (get-in @db [:regions @current-region-id :geojson])))))
+
+(register-sub
+ :current-project/view-state
+ (fn [db [_]]
+   (reaction (get-in @db [:current-project :view-state]))))
+
+(register-sub
+ :current-project/sharing-emails-text
+ (fn [db [_]]
+   (reaction (get-in @db [:current-project :sharing :emails-text]))))
+
+(register-sub
+ :current-project/sharing-send-emails
+ (fn [db [_ field]]
+   (let [emails-text (subscribe [:current-project/sharing-emails-text])]
+     (reaction
+       (let [emails-list    (db/split-emails @emails-text)
+             state          (get-in @db [:current-project :sharing :state])
+             has-emails     (seq emails-list)
+             valid-emails   (filter utils/is-valid-email? emails-list)
+             invalid-emails (filter (complement utils/is-valid-email?) emails-list)]
+         {:editable (not (#{:sending} state))
+          :disabled (or (#{:sending :sent} state) (not has-emails) (seq invalid-emails))
+          :valid    (not (seq invalid-emails))
+          :title    (cond
+                      (= :sending state)   "Sending..."
+                      (= :sent state)      "Emails successfully sent"
+                      (not has-emails)     "Enter one or more email addresses"
+                      (seq invalid-emails) (str
+                                              (string/join ", " invalid-emails) " "
+                                              (if (= 1 (count invalid-emails))
+                                                "is an invalid email address"
+                                                "are invalid email addresses"))
+                      :else                "Send this project sharing link to the emails above")
+          :label    (cond
+                      (= :sending state) "Sending..."
+                      (= :sent state)    "Sent"
+                      (seq valid-emails) (str "Send to " (utils/pluralize (count valid-emails) "user"))
+                      :else              "Send")})))))
```

Comment:
There is some view-only related code here that I think belongs in a Reagent component. Eg. title and label, probably editable and disabled as well. Otherwise the `share-dialog` component is tightly coupled to this subscription for no particular reason.